index.js

```diff
@@ -99,7 +94,60 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {},
+  callback,
+  callbackOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  let response
+  try {
+    response = await query(pointer)
+  } catch (err) {
+    const endpoint = new URL(pointer.startsWith('$')
+      ? 'https://' + pointer.substring(1)
+      : pointer)
+    const postResponse = await fetch(endpoint.href, {
+      method: 'post'
+    })
+    if (postResponse.status !== 200) {
+      throw new Error('got error response from spsp server.' +
+        ' endpoint="' + endpoint.href + '"' +
+        ' status=' + postResponse.status +
+        ' message="' + (await postResponse.text()) + '"')
+    }
+  }
+  response = await query(pointer)
```

Comment:
This could be moved inside the `catch` block

Reply:
Yes, I'll move it.